mycore-base/src/main/java/org/mycore/frontend/support/MCRAutoDeploy.java

```diff
@@ -123,107 +123,116 @@ private void deployWebResources(final ServletContext servletContext, final MCRCo
+                                                }
+                                            }).orElse(() -> LOGGER
+                                                .warn("Filter " + name + " already registered!"))
+                                            .run();
+                                    });
+                            });
+
+                            final List<Element> servlets = root.getChildren("servlet", ns);
+                            final List<Element> smaps = root.getChildren("servlet-mapping", ns);
+
+                            servlets.forEach(servlet -> {
+                                final String name = servlet.getChildText("servlet-name", ns);
+                                final String className = servlet.getChildText("servlet-class", ns);
+
+                                smaps.stream()
+                                    .filter(mapping -> mapping.getChildText("servlet-name", ns).equals(name))
+                                    .findFirst()
+                                    .ifPresent(mapping -> {
+                                        LOGGER.info("Register Servlet " + name + " (" + className + ")...");
+                                        Optional.ofNullable(servletContext.addServlet(name, className))
+                                            .<Runnable> map(sr -> () -> {
+                                                mapping.getChildren("url-pattern", ns).stream()
+                                                    .forEach(url -> {
+                                                        LOGGER.info("...add url mapping: " + url.getTextTrim());
+                                                        sr.addMapping(url.getTextTrim());
+                                                    });
+                                            }).orElse(() -> LOGGER
+                                                .error("Servlet" + name + " already registered!"))
+                                            .run();
+                                    });
+                            });
+                        } catch (IOException | JDOMException e) {
+                            LOGGER.error("Couldn't parse " + WEB_FRAGMENT, e);
+                        }
+                    });
 
-            jar.close();
-        } catch (final IOException e) {
-            LOGGER.error("Couldn't parse JAR!", e);
+                jar.close();
+            } catch (final IOException e) {
+                LOGGER.error("Couldn't parse JAR!", e);
+            }
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private boolean isHandledByServletContainer(ServletContext servletContext, MCRComponent comp) {
+        List<String> orderedLibs = (List<String>) servletContext.getAttribute(ServletContext.ORDERED_LIBS);
+        return orderedLibs.stream().filter(s -> s.equals(comp.getJarFile().getName())).count() != 0;
```

Comment:
`stream.filter().anyMatch(s -> s.equals(comp.getJarFile().getName()))` is using short-circuiting of the stream API